### DISCUSSION TITLE
eror when try to add identity_providers: to my authelia docker

### DISCUSSION THREAD

**Discussion by ghsighsi** [9p4/jellyfin-plugin-sso#173] [Q&A]
hey guys im trying to add the sso to my jellyfin server and i cant run my authelia for some reason 

2024/03/25 15:23:20 | stdout | [31mFATA[0m[0000] Can't continue due to the errors loading the configuration
2024/03/25 15:23:20 | stdout | [31mERRO[0m[0000] Configuration: identity_providers: oidc: option 'issuer_private_key' is required

when I add the client block the the config, that the log that pop.
some one know why?

i used this block and filled it with my details

identity_providers:
oidc:
// hmac secret and private key given by env variables
clients:
- id: jellyfin
description: My media server
// Client secret should be randomly generated
secret:
authorization_policy: one_factor
redirect_uris:
- https://jellyfin.example.com/sso/OID/redirect/authelia

**Comment by 9p4** [comment]
Please refer to https://www.authelia.com/reference/guides/generating-secure-values/#generating-an-rsa-keypair

**Reply by ghsighsi** [reply]
Hey 9p4
I new to this, I'm not supposed to add it according to the git guide. If I do where I add this to the block?

**Reply by 9p4** [reply]
My guide assumes that you already have Authelia installed properly and running. Please properly set up and configure Authelia before attempting to use my plugin. Please refer to [the Authelia getting started guide](https://www.authelia.com/integration/prologue/get-started), and follow the instructions there. Please ensure that you apply the suggestions in the section "Moving to Production" for security reasons.

**Reply by ghsighsi** [reply]
i set a working authelia docker but when i had the oidc configure the eror shown and its wont start, but when i add issuer_private_key line it all crush and i get 30 erors

**Reply by 9p4** [reply]
Can you upload your entire Authentik configuration (with secret values redacted) here in a code block?

**Reply by ghsighsi** [reply]
###############################################################################
                           Authelia Configuration                            #
###############################################################################

theme: dark #dark
jwt_secret: <redact> #any text or number you want to add here to create jwt Token

default_redirection_url: https://google.com/ #where to redirect for a non-existent URL

server:
  host: 192.168.0.107
  port: 9091
  path: ""
  enable_pprof: false
  enable_expvars: false
  disable_healthcheck: false
  tls:
    key: ""
    certificate: ""

log:
  level: debug

totp:
  issuer: <redact> #your authelia top-level domain
  period: 30
  skew: 1

duo_api:
  disable: false
  hostname: '<redact>'
  integration_key: '<redact>'
  secret_key: '<redact>'
  enable_self_enrollment: false

authentication_backend:
  refresh_interval: 5m
  file:
    path: /config/users_database.yml #this is where your authorized users are stored
    password:
      algorithm: argon2id
      iterations: 1
      key_length: 32
      salt_length: 16
      memory: 1024
      parallelism: 8

access_control:
  default_policy: deny
  rules:
    ## bypass rule
    - domain: 
        - "<redact>" #This should be your authentication URL
      policy: bypass
    - domain: "<redact>" #example domain to protect
      policy: two_factor


session:
  name: authelia_session
  secret: <redact> #any text or number you want to add here to create jwt Token
  expiration: 14400  # 4 hour
  inactivity: 3600  # 60 minutes
  domain: <redact>  # Should match whatever your root protected domain is

regulation:
  max_retries: 3
  find_time: 10m
  ban_time: 30m

storage:
  local:
    path: /config/db.sqlite3 #this is your databse.
  encryption_key: "<redact>"

identity_providers:
  oidc:
      clients:
      - id: jellyfin
        description: My media server
        secret: <redact>
        authorization_policy: two_factor
        redirect_uris:
          - https://jelly.com/sso/OID/redirect/authelia
  
notifier:
  disable_startup_check: true #true/false
  smtp:
    username: <redact> #your email address
    password: <redact> #your email password
    host: smtp.gmail.com #email smtp server
    port: 587 #email smtp port
    sender: <redact>
    identifier: localhost
    subject: "[Authelia] {title}" #email subject
    startup_check_address: <redact>
    disable_require_tls: false
    disable_html_emails: false
    tls:
      skip_verify: false
      minimum_version: TLS1.2
...



thats my authelia conf yml, you see somthing missing?

**Reply by 9p4** [reply]
Can you upload the authelia config properly formatted? ie, surrounded by backticks like so:

````
```
yaml goes here
```
````

**Reply by taiwan-king** [reply]
> ###############################################################################Authelia 配置# ###############################################################################
> 
> 主題：深色 #dark jwt_secret：#any 要在此處添加以創建 jwt 令牌的文本或數位
> 
> default_redirection_url：https://google.com/#where 重定向到不存在的 URL
> 
> 伺服器： 主機： 192.168.0.107 埠： 9091 路徑： “” enable_pprof： false enable_expvars： false disable_healthcheck： false TLS： 密鑰： “” 憑證： “”
> 
> 日誌：級別：調試
> 
> TOTP： 頒發者： #your Authelia 頂級域 週期： 30 偏斜： 1
> 
> duo_api： 禁用： false 主機名： '' integration_key： '' secret_key： '' enable_self_enrollment： false
> 
> authentication_backend：refresh_interval：5M 檔：路徑：/config/users_database.yml #this 是您的授權使用者的存儲位置 密碼：演算法：argon2id 迭代次數：1 key_length：32 salt_length：16 記憶體：1024 並行度：8
> 
> access_control： default_policy： 拒絕規則： ## 繞過規則 - 域： - “” #This 應該是您的身份驗證 URL 策略： 繞過 - 域： “” #example 域以保護策略： two_factor
> 
> session： name： authelia_session secret： #any 你想在此處添加以創建 jwt 的文本或數位 令牌過期時間： 14400 # 4 小時不啟用時間： 3600 # 60 分鐘 domain： # 應該匹配你的根保護域
> 
> 調節： max_retries： 3 find_time： 10m ban_time： 30m
> 
> storage： local： path： /config/db.sqlite3 #this 是您的資料庫。 encryption_key：“”
> 
> identity_providers： oidc： clients： - id： jellyfin 描述： 我的媒體伺服器密鑰： authorization_policy two_factor redirect_uris： - https://jelly.com/sso/OID/redirect/authelia
> 
> 通知符： disable_startup_check： true #true/false smtp： 使用者名： #your 電子郵件地址 密碼： #your 電子郵件密碼 主機： smtp.gmail.com #email SMTP 伺服器埠： 587 #email SMTP 埠 發件者： 標識符： localhost 主題： “[Authelia] {title}” #email 主題 startup_check_address： disable_require_tls： false disable_html_emails： falsetls： skip_verify： false minimum_version： TLS1.2 ...
> 
> 那是我的 Authelia conf yml，你看到少了什麼嗎？

Your configuration does not follow the official standard examples at all. If you check the authelia official documentation, you'll realize you're missing a lot of settings.
